qutebrowser/browser/shared.py

```diff
@@ -310,7 +310,12 @@ def netrc_authentication(url, authenticator):
     user, password = None, None
     try:
         net = netrc.netrc(config.val.content.netrc_file)
-        authenticators = net.authenticators(url.host())
+        if url.port() != -1:
+            authenticators = net.authenticators(
+                "{}:{}".format(url.host(), url.port())
+            )
+        if not authenticators:
```

Comment:
This actually was broken, because `authenticators` is never set if the port is -1:

```
Traceback (most recent call last):
  File "/home/florian/proj/qutebrowser/git/qutebrowser/browser/webengine/webenginetab.py", line 1307, in _on_authentication_required
    netrc_success = shared.netrc_authentication(url, authenticator)
  File "/home/florian/proj/qutebrowser/git/qutebrowser/browser/shared.py", line 317, in netrc_authentication
    if not authenticators:
UnboundLocalError: local variable 'authenticators' referenced before assignment
```

Fixed in 73bd88b55ad48e0ae69f6af57e2230b99c6d60fe.